validationparser/interceptor_test.go

```diff
@@ -0,0 +1,208 @@
+package validationparser
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/transport"
+)
+
+// DummyContextWithServerTransportStream returns a dummy context for testing.
+func DummyContextWithServerTransportStream() context.Context {
+	expectedStream := &transport.Stream{}
+	return grpc.NewContextWithServerTransportStream(context.Background(), expectedStream)
+}
+
+// TestUnaryServerInterceptor_ValidationErrors will run mock validation errors to see if it parses correctly.
+func TestUnaryServerInterceptor_ValidationErrors(t *testing.T) {
+	ctx := DummyContextWithServerTransportStream()
+	tests := []struct {
+		name     string
+		want     mockRequestValidationError
+		expected error
```

Comment:
`want` and `expected` are synonymous in the testing realm... what is meant by these?

My assumption is that `want` -> `input` and `expected` -> `want`